cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -41,20 +41,16 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
     match op {
         Opcode::Iconst | Opcode::Bconst | Opcode::Null => {
             let value = ctx.get_constant(insn).unwrap();
```

Comment:
`get_constant` doesn't sign extend the immediate. It returns the raw bits. cg_clif chooses to zero extend immediates. https://github.com/bytecodealliance/wasmtime/blob/bae4ec642798ff448ca88eab771b6fcea71e7884/cranelift/codegen/src/inst_predicates.rs#L60